dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -1360,10 +1373,19 @@ private String getFromSubQueryLimitAndOffset( TrackedEntityInstanceQueryParams p
     {
         StringBuilder limitOffset = new StringBuilder();
         int limit = params.getMaxTeiLimit();
+        int teiQueryLimit = systemSettingManager.getIntSetting( SettingKey.TRACKED_ENTITY_MAX_LIMIT );
 
         if ( limit == 0 && !params.isPaging() )
         {
-            return "";
+
+            return Optional.of( teiQueryLimit ).filter( a -> teiQueryLimit == 0 ).map( a -> "" )
```

Comment:
I think we should simplify this into

```java
        if (teiQueryLimit > 0) {
            limitOffset
                    .append("limit")
                    .append(" ")
                    .append(teiQueryLimit)
                    .append(" ");
        }

        return limitOffset.toString();
```

I also assume a limit `teiQueryLimit < 0` does not make sense hence the `teiQueryLimit > 0` check.

Reply:
@zubaira can you please address this 😄

Reply:
resolved